hawc/apps/animal/templates/animal/animalgroup_detail.html

```diff
@@ -1,32 +1,33 @@
 {% extends 'assessment-rooted.html' %}
 
 {% load hawc %}
+{% load bs4 %}
 
 {% block content %}
     {% if obj_perms.edit %}
-      <div class="dropdown btn-group float-right">
-        <a class="btn btn-primary dropdown-toggle" data-toggle="dropdown">Actions</a>
-        <div class="dropdown-menu dropdown-menu-right">
-            <span class="dropdown-header">Animal Group Editing</span>
-            <a class="dropdown-item" href="{% url 'animal:animal_group_update' object.pk %}">Update</a>
-            <a class="dropdown-item" href="{% audit_url object %}">Change logs</a>
-            <a class="dropdown-item" href="{% url 'animal:animal_group_delete' object.pk %}">Delete</a>
-
-            {% if object.dosing_regime and object == object.dosing_regime.dosed_animals %}
+    <div class="d-flex">
+      <h2 class="d-inline-block mb-0">{{object}}</h2>
+      {% actions %}
+              <span class="dropdown-header">Animal Group Editing</span>
+              <a class="dropdown-item" href="{% url 'animal:animal_group_update' object.pk %}">Update</a>
+              <a class="dropdown-item" href="{% audit_url object %}">Change logs</a>
+              <a class="dropdown-item" href="{% url 'animal:animal_group_delete' object.pk %}">Delete</a>
+
+              {% if object.dosing_regime and object == object.dosing_regime.dosed_animals %}
+                <div class="dropdown-divider"></div>
+                <span class="dropdown-header">Dose Regime Editing</span>
+                <a class="dropdown-item" href="{% url 'animal:dosing_regime_update' object.dosing_regime.pk %}">Update</a>
+                <a class="dropdown-item" href="{% audit_url object.dosing_regime %}">Change logs</a>
+              {% endif %}
+
               <div class="dropdown-divider"></div>
-              <span class="dropdown-header">Dose Regime Editing</span>
-              <a class="dropdown-item" href="{% url 'animal:dosing_regime_update' object.dosing_regime.pk %}">Update</a>
-              <a class="dropdown-item" href="{% audit_url object.dosing_regime %}">Change logs</a>
-            {% endif %}
-
-            <div class="dropdown-divider"></div>
-            <span class="dropdown-header">Endpoint Editing</span>
-            <a class="dropdown-item" href="{% url 'animal:endpoint_new' object.pk %}">Create new</a>
-            <a class="dropdown-item" href="{% url 'animal:endpoint_copy_selector' object.pk %}">Copy from existing</a>
-        </div>
-      </div>
-    {% endif %}
-    <h2 id="animal-group-header" class="d-inline-block">{{object}}</h2>
+              <span class="dropdown-header">Endpoint Editing</span>
+              <a class="dropdown-item" href="{% url 'animal:endpoint_new' object.pk %}">Create new</a>
+              <a class="dropdown-item" href="{% url 'animal:endpoint_copy_selector' object.pk %}">Copy from existing</a>
+              {% endactions %}
+    </div>
+      <h2 id="animal-group-header" class="mt-0"></h2>
```

Comment:
maybe always put the header before the actions button, since that makes sense reading left to right? is this second header needed? if it is, let me know and we can fix the JS

Reply:
I believe its a js problem. Youll notice there are two h2's in the template. If i put the 'animal-group-header' id in the h2 before the actions button, the details table and dosing regime will render in between the header and actions button and mess up the page

Reply:
no problem. just add a `TODO` and we can come back to it w/ the final review